src/Algebra/Consequences/Base.agda

```diff
@@ -13,14 +13,48 @@ module Algebra.Consequences.Base
 open import Algebra.Core
 open import Algebra.Definitions
 open import Data.Sum.Base
+open import Function.Base using (const)
 open import Relation.Binary.Core
 open import Relation.Binary.Definitions using (Reflexive)
+open import Relation.Nullary.Decidable.Core using (yes; no)
+open import Relation.Nullary.Negation.Core using (contradiction)
+open import Relation.Unary using (Pred; Decidable)
+
 
 module _ {ℓ} {_•_ : Op₂ A} (_≈_ : Rel A ℓ) where
 
   sel⇒idem : Selective _≈_ _•_ → Idempotent _≈_ _•_
   sel⇒idem sel x = reduce (sel x x)
 
+module _ {p ℓ} {_•_ : Op₂ A} (_≈_ : Rel A ℓ) (P : Pred A p) where
+
+  almost⇒exceptˡ : _-AlmostLeftCancellative_ _≈_ P _•_ →
+                   Except_-LeftCancellative_ _≈_ P _•_
+  almost⇒exceptˡ cancelˡ {x} with cancelˡ x
+  ... | inj₁ px     = contradiction px
+  ... | inj₂ cancel = const cancel
+
+  almost⇒exceptʳ : _-AlmostRightCancellative_ _≈_ P _•_ →
+                   Except_-RightCancellative_ _≈_ P _•_
+  almost⇒exceptʳ cancelʳ {x} with cancelʳ x
+  ... | inj₁ px     = contradiction px
+  ... | inj₂ cancel = const cancel
```

Comment:
Moderately sure that @JacquesCarette would prefer a definition in terms of `Data.Sum.Base.[_,_]` ;-)
Next iteration...